readme.md

```diff
@@ -0,0 +1,161 @@
+# Remark for Visual Studio Code
+
+[![Build][build-badge]][build]
+[![Coverage][coverage-badge]][coverage]
+[![Downloads][downloads-badge]][downloads]
+[![Sponsors][sponsors-badge]][collective]
+[![Backers][backers-badge]][collective]
+[![Chat][chat-badge]][chat]
+
+**[remark][]** is a tool that transforms markdown with plugins.
+These plugins can inspect and change your markup.
+You can use remark on the server, the client, CLIs, deno, etc.
+This plugin for Visual Studio Code lets you format and lint markdown files using
+remark.
+
+## Contents
+
+*   [When should I use this?](#when-should-i-use-this)
+*   [Use](#use)
+    *   [Formatting](#formatting)
+*   [Plugins](#plugins)
+*   [Syntax](#syntax)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## When should I use this?
+
+If your project is checked using [`remark-cli`][remark-cli] and you use Visual
+Studio Code to edit markdown files, it’s recommended to use this extension.
+
+## Use
+
+To start using this for a project, [`remark-cli`][remark-cli] needs to be
+configured first.
+For example, install `remark-cli` and `remark-preset-lint-recommended`:
+
+```sh
+npm install remark-cli remark-preset-lint-recommended
+```
+
+Now create a file named `.remarkrc.json` with the following content:
+
+```json
+{
+  "plugins": [["preset-lint-recommended"]]
+}
+```
+
+Now any syntax which doesn’t comply with `remark-preset-lint-recommended` will
+be reported in the *Problems* pane in your editor.
+
+### Formatting
+
+This extension can be used as a formatter for markdown files.
+
+To format a file, first press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>.
+Search for and select *“Format Document With…”*.
+Now select *“Remark”*.
+
+To make this the default formatter, add the following to your `settings.json`
+(Open with <kbd>Ctrl</kbd>+<kbd>,</kbd>):
+
+```json
+{
+  "[markdown]": {
+    "editor.defaultFormatter": "unifiedjs.vscode-remark",
+  }
+}
+```
+
+Optionally enable format on save:
+
+```json
+{
+  "[markdown]": {
+    "editor.defaultFormatter": "unifiedjs.vscode-remark",
+    "editor.formatOnSave": true
+  }
+}
+```
+
+## Plugins
+
+The **remark** ecosystem has a variety of plugins available.
+Most notably you’ll want to check out [`remark-lint`][remark-lint].
+A curated list of plugins can be found [here][list-of-plugins].
+
+## Syntax
+
+remark follows CommonMark, which standardizes the differences between markdown
+implementations, by default.
+Some syntax extensions are supported through plugins.
+
+## Compatibility
+
+Projects maintained by the unified collective are compatible with all maintained
+versions of Node.js.
+As of now, that is Node.js 12.20+, 14.14+, and 16.0+.
```

Comment:
Is this compat range useful for VS Code users? Should this be about VS Code support?

Reply:
I’m not entirely sure. `remark-language-server` is bundled with the extension. The NodeJS support ranges do apply to this.

Also I don’t really know how VSCode handles compatibility for extensions.

Reply:
VS Code does bundle Node though, so Node.js doesn‘t matter, but VS Code should?
Also: isn’t esbuild used?

Reply:
The `node` executable is used to run the language server. I don’t think VS Code injects its own `PATH`.

That’s what’s nice about using a language server: it doesn’t depend on the editor. Both happen to use Node.js, but that’s a coincidence.

I wouldn’t rely on esbuild for compatibility. This can likely fix syntactic compatibility issues, but not for example new methods introduced in Node.js 12.20+, which unified may depend on.

---

Looking at the answer I just typed I think the answer is simply: Yes, the Node.js compatibility range is useful.

Reply:
OK, sounds good.
I feel like it would make sense to include a VSC version of it applies and is findable someway

Reply:
I documented it works with VS Code 1.46.0 and greater, because that’s what’s documented in the `package.json` engines field. I don’t know what this means in practice. I think most people update VS Code regularly anyway. :shrug:

Reply:
Hmm, so the `engines` field is likely wrong as well? :(

Reply:
It’s probably fine. I’m guessing bootstrapping a new project simply sets it to the current latest version of VS Code.